src/main/java/Deadline.java

```diff
@@ -0,0 +1,41 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Deadline extends Task {
+
+    protected String by;
+    String dateAsString = "";
```

Comment:
Instead of storing formatted strings as properties, you can have static formatters as methods to perform this formatting. That would allow you to change the formatting in the future by editing only a single method or add more formatters as required without affecting the properties of the Class.